ui/ui-components/pages/_app.tsx

```diff
@@ -102,6 +103,17 @@ GrouparooNextApp.getInitialProps = async (appContext: AppContext) => {
     try {
       appProps = await App.getInitialProps(appContext);
     } catch (_error) {
+      const { req, res } = appContext.ctx;
+      const requestPath = req.url.match("^[^?]*")[0];
+      const redirect = getRedirectFromErrorCode(_error.code, requestPath);
+      if (redirect) {
+        res.writeHead(302, {
+          Location: redirect.destination,
+        });
+        res.end();
+        return {};
+      }
+
```

Comment:
This needs to exist as long as we have uses `getInitialProps`. There are 58 files that currently use `getInitialProps`. I could create a separate change to migrate them all to `getServerSideProps`

Reply:
Ok, But I will keep the change as an isolated PR. For now I'll move the support here and remove this stuff once in the migration PR